app/grandchallenge/reader_studies/forms.py

```diff
@@ -310,6 +322,54 @@ def interface_choices(self):
     def initial_interface(self):
         return self.interface_choices().first()
 
+    def widget_choices(self):
+        answer_type = self["answer_type"].value()
+        choices = [("", "Default")]
+        if answer_type:
+            choices.extend(
+                QuestionWidget.get_widget_choices_for_answer_type(
+                    answer_type=answer_type
+                )
+            )
+        return choices
+
+    def initial_widget(self):
+        try:
+            return self.instance.widget.kind
+        except ObjectDoesNotExist:
+            return ""
+
+    def clean_widget(self):
+        value = self.cleaned_data.get("widget")
+        if value == "":
+            try:
+                self.instance.widget.delete()
```

Comment:
Should you be deleting objects in clean? I would have thought that should only be done on save?

Reply:
You're totally right. I'm working on handling the widget saving and deleting in the save method now, much cleaner.